nxcomp/src/Loop.cpp

```diff
@@ -14731,7 +14731,19 @@ static void handleAlertInLoop()
     {
       char caption[DEFAULT_STRING_LENGTH];
 
-      strcpy(caption, ALERT_CAPTION_PREFIX);
+      const char* caption_prefix = getenv("NX_DIALOG_CAPTIONPREFIX");
+
+      if (!caption_prefix) {
+          // Use the hard-coded 'NX -' as default.
+          strncpy(caption, ALERT_CAPTION_PREFIX, DEFAULT_STRING_LENGTH);
+          caption[DEFAULT_STRING_LENGTH-1] = '\0';
+      }
+      else
+      {
+          // Use what gets provided via NX_DIALOG_CAPTIONPREFIX (used in X2Go)
+          strncpy(caption, caption_prefix, DEFAULT_STRING_LENGTH-1);
```

Comment:
this strncpy is different from the one above regarding the number of characters to be copied. While both are technically correct they should alt least be unified.

Reply:
Can you suggest a clean solution? (You are the C dev... ;-) ).

Reply:
str(n)cpy is always tricky. I would do the getenv only once. Then strdup either the result or the default. And free the dupped string at shutdown.

Reply:
@uli42 Can you help with this MR and fine-polish it? Your C is so much better than mine.